src/AI-Algorithms-Graph/AIDijkstra.class.st

```diff
@@ -89,9 +85,10 @@ AIDijkstra >> reset [
 
 { #category : #running }
 AIDijkstra >> run [
-
+	
 	| pq |
 	pq := self newPriorityQueue.
+	pq sortBlock: [ :element1 :element2 | (element1 value) <= (element2 value)].
 	pq add: start -> 0.
```

Comment:
with the first implementation we were adding the elements to the priority queue as an association `pq add: start -> 0`. The second parameter, the `0`, was the priority. That was after used here `minValue := aPriorityQueue detectMin: [ :assoc | assoc value ].`.

So, as we are know using the Heap then we don't need anymore to use that association nor define the sorting block

Reply:
I have also removed the association and made the algorithm work by adding a priority attribute to the node class, as we had discussed.